pkg/graveler/graveler.go

```diff
@@ -28,6 +30,37 @@ const (
 	DiffTypeConflict
 )
 
+type RefModType rune
+
+const (
+	RefModTypeTilde  RefModType = '~'
+	RefModTypeCaret  RefModType = '^'
+	RefModTypeAt     RefModType = '@'
+	RefModTypeDollar RefModType = '$'
+)
+
+type RefModifier struct {
+	Type  RefModType
+	Value int
+}
+
+type ParsedRef struct {
+	BaseRef      string
+	RefModifiers []RefModifier
+}
+
+func (p *ParsedRef) Ref() Ref {
```

Comment:
Missing documentation

Reply:
will remove this one - thought it will be required for a test